### ISSUE TITLE
x-axis changed with --loglength

### ISSUE THREAD

**Issue by GeoMicroSoares** [wdecoster/NanoPlot#15]
Hello again,

Here's what I get running the default NanoPlot command with my Phred score >9 .fastq.
![bp_v1_qf9_nanoplot](https://user-images.githubusercontent.com/14272488/31662626-18cdd23c-b337-11e7-8cd9-562625d204f3.png)

Here's what I get when I run it with --loglength:
![bp_v1_qf9_nanoplot_log](https://user-images.githubusercontent.com/14272488/31662655-38102370-b337-11e7-9c31-4816816a02f9.png)

And the .log:
```
$ cat BP_v1_qf9_log_dot/BP_v1_qf9_log_dot_Nanoplot_20171013_1157.log
2017-10-13 11:57:44,606 Nanoplot 0.18.0 started with arguments Namespace(alength=False, bam=None, color='#4CB391', downsample=None, drop_outliers=False, fastq=['nanoplot_BP_v1_qf9/BP_v1_reBC_qf9.fastq'], fastq_rich=None, format='png', listcolors=False, loglength=True, maxlength=None, outdir='./BP_v1_qf9_log_dot/', plots=['dot'], prefix='BP_v1_qf9_log_dot_', readtype='1D', summary=None, threads=8)
2017-10-13 11:57:44,606 Python version is: 3.4.3 (default, Nov 17 2016, 01:08:31)  [GCC 4.8.4]
2017-10-13 11:57:44,606 Versions of key modules are:
2017-10-13 11:57:44,606 <module 'numpy' from '/usr/local/lib/python3.4/dist-packages/numpy/__init__.py'>: 1.13.1
2017-10-13 11:57:44,608 <module 'seaborn' from '/usr/local/lib/python3.4/dist-packages/seaborn/__init__.py'>: 0.8.1
2017-10-13 11:57:44,608 <module 'pandas' from '/usr/local/lib/python3.4/dist-packages/pandas/__init__.py'>: 0.20.3
2017-10-13 11:57:44,608 <module 'pysam' from '/usr/local/lib/python3.4/dist-packages/pysam/__init__.py'>: 0.12.0.1
2017-10-13 11:57:44,609 <module 'matplotlib' from '/usr/local/lib/python3.4/dist-packages/matplotlib/__init__.py'>: 2.1.0
2017-10-13 11:57:44,609 <module 'nanoget' from '/usr/local/lib/python3.4/dist-packages/nanoget/__init__.py'>: 0.14.1
2017-10-13 11:57:44,609 <module 'nanoplotter' from '/usr/local/lib/python3.4/dist-packages/nanoplotter/__init__.py'>: 0.14.1
2017-10-13 11:57:44,609 <module 'nanomath' from '/usr/local/lib/python3.4/dist-packages/nanomath/__init__.py'>: 0.13.2
2017-10-13 11:57:44,609 Nanoplotter: valid output format png
2017-10-13 11:57:44,609 Nanoget: Starting to collect statistics from plain fastq file.
2017-10-13 11:57:44,949 Nanoget: Finished collecting statistics from plain fastq file.
2017-10-13 11:57:44,951 Gathered metrics for plotting
2017-10-13 11:57:44,953 Calculated statistics
2017-10-13 11:57:44,953 Using sequenced read lengths for plotting.
2017-10-13 11:57:44,953 Using Log10 scaled read lengths.
2017-10-13 11:57:44,954 Processed the metrics, optionally performed filtering.
2017-10-13 11:57:44,954 Nanoplotter: Valid color #4CB391.
2017-10-13 11:57:44,954 Nanoplotter: Creating length plots for Read length.
2017-10-13 11:57:44,954 Nanoplotter: Using 210 reads with read length N50 of 541.
2017-10-13 11:57:45,550 Created length plots
2017-10-13 11:57:45,550 Nanoplotter: Creating Read lengths vs Average read quality plots using statistics from 210 reads.
2017-10-13 11:57:45,922 Created LengthvsQual plot
2017-10-13 11:57:45,923 Succesfully processed all input.
```
Thanks again!

**Comment by wdecoster** [comment]
Hi,
Thanks for both reports, I will look at these issues tomorrow since I am at the Utrecht Nanopore meeting today. 

Cheers, 
Wouter

**Comment by wdecoster** [comment]
Hi,

Just to avoid this is an issue due to an outdated version, could you update the `nanoplotter` and `nanomath` submodules and replicate the issue?
`pip install nanoplotter -U`  
`pip install nanomath -U`

Thanks!

**Comment by wdecoster** [comment]
Hi,

I just wrote fastq-anonymous (https://github.com/wdecoster/fastq-anonymous).  
The purpose of this script is to anonymize the sequence and fastq identifier - while keeping read lengths and qualities intact. As such you can share your input file which creates this unexpected result, allowing me to replicate the issue and troubleshoot. 

Cheers,  
Wouter

**Comment by GeoMicroSoares** [comment]
Heya @wdecoster 

Updating the modules didn't work, unfortunately.

Tried to install fastq-anonymous but got:
```
$ pip install fastq-anonymous
Collecting fastq-anonymous
  Downloading fastq-anonymous-1.0.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-o_6f57yo/fastq-anonymous/setup.py", line 8, in <module>
        exec(open('fastqanonymous/version.py').read())
      File "/usr/lib/python3.4/codecs.py", line 891, in open
        file = builtins.open(filename, mode, buffering)
    FileNotFoundError: [Errno 2] No such file or directory: 'fastqanonymous/version.py'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-o_6f57yo/fastq-anonymous/
```

Maybe it hasn't been completely uploaded to github yet.

**Comment by wdecoster** [comment]
Woop - I made a small mistake there.  
Should be fixed in v1.0.1, so if you could try reinstalling...

You might need
`pip install fastq-anonymous --no-cache-dir`

**Comment by GeoMicroSoares** [comment]
@wdecoster here is the file generating the problems. Funnily enough this is the only file not working...
[anonymous_reads.zip](https://github.com/wdecoster/NanoPlot/files/1397729/anonymous_reads.zip)
Thanks again!

**Comment by wdecoster** [comment]
Hi,

Thanks for sharing your file. That will make troubleshooting easier :)
I just used the anonymized file and for the default plot I get the following:

![default-notreproduced](https://user-images.githubusercontent.com/11660522/31764102-8731bc18-b4c0-11e7-8b28-fcc62598af23.png)

That looks pretty normal to me, except that the Y-axis (and quality scores) are not the same as what you showed at the top. My first guess would be that this is due to an outdated nanomath submodule in which the quality scores were incorrectly averaged.

When I do the loglength plot I see the x-axis which is not okay - so that's something I definitely have to fix.

Cheers,  
Wouter

**Comment by wdecoster** [comment]
Fixed in nanoplotter 0.16.4, thanks for your patience!

If you could confirm the x-axis looks as expected that would be great!
